### PR TITLE
tests: use UTF-8 when creating metrics files

### LINKED ISSUE #1929: test: TestMetrics.test_formatted_output may fail on Windows
It runs successfully on appveyor, but fails on Windows 10 free image with:

> E       AssertionError: assert '\tmetrics.csv:\n\t\tvalor_mse   desviación_mse   data_set       \n\t\t0.421601    0.173461         entrenamiento  \n\t\t0.67528     0.289545         pruebas        \n\t\t0.671502    0.297848         validación' in 'unable to read metric in \'metrics.csv\' in branch \'\'\n\tmetrics.tsv:\n\t\tvalue_mse   deviation_mse   data_set    ...\t\t          "0.671502"\n\t\t     ]\n\t\t}\n\tmetrics.txt:\n\t\tROC_AUC: 0.64\n\t\tKS: 78.9999999996\n\t\tF_SCORE: 77'

Note that "unable to read metric in 'metrics.csv' in branch ...", the full error is:

```
 unable to read metric in 'metrics.csv' in branch ''
Traceback (most recent call last):
  File "E:\dvc\repo\metrics\show.py", line 138, in _read_metric
    return _format_output(fd.read().strip(), typ)
  File "c:\users\user\envs\dvc\lib\codecs.py", line 322, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xf3 in position 18: invalid continuation byte
```

So it's something about encodongs. Most probably some `open()` somewhere doesn't specify encoding and that falls back to system default, say cp1252, which somewhat works for text in the test, but fails later then it's read as unicode.

### PR DESCRIPTION
**Pull request by mroutis**
@efiop , @Suor, this issue is going to "fix" only the faulty test, let's discuss all the encoding related topics on a [different issue](https://github.com/iterative/dvc/issues/1957) and take a decision about how we should manage this across the code base.

* [x] Have you followed the guidelines in our
      [Contributing document](https://dvc.org/doc/user-guide/contributing)?

* [x] Does your PR affect documented changes or does it add new functionality
      that should be documented? If yes, have you created a PR for
      [dvc.org](https://github.com/iterative/dvc.org) documenting it or at
      least opened an issue for it? If so, please add a link to it.

-----

Fix #1929

### CODE REVIEW

File: dvc/scm/git/tree.py [hunk 1]
```diff
@@ -30,7 +30,6 @@ def tree_root(self):
         return self.git.working_dir
 
     def open(self, path, binary=False):
-
```
Comment (efiop) [inline dvc/scm/git/tree.py]: How are these whitelines related to the patch? Please avoid polluting your commits with unrelated changes.

Comment (mroutis) [reply dvc/scm/git/tree.py]: not related at all, my mistake.
